src/EbmlUnicodeString.cpp

```diff
@@ -151,21 +151,15 @@ filepos_t EbmlUnicodeString::RenderData(IOCallback & output, bool /* bForceRende
 
 EbmlUnicodeString::operator const UTFstring &() const {return Value;}
 
-EbmlUnicodeString & EbmlUnicodeString::operator=(const UTFstring & NewString)
-{
-  Value = NewString;
+EbmlUnicodeString &EbmlUnicodeString::SetValue(UTFstring const &NewValue) {
+  Value = NewValue;
   SetValueIsSet();
   return *this;
 }
 
-EbmlUnicodeString &EbmlUnicodeString::SetValue(UTFstring const &NewValue) {
-  return *this = NewValue;
-}
-
 EbmlUnicodeString &EbmlUnicodeString::SetValueUTF8(std::string const &NewValue) {
-  UTFstring NewValueUTFstring;
-  NewValueUTFstring.SetUTF8(NewValue);
-  return *this = NewValueUTFstring;
+  Value.SetUTF8(NewValue);
```

Comment:
Here a `SetValueIsSet()` is missing. The result is an assertion in `UpdateSize` that `CheckMandatory` failed. Almost all tests in MKVToolNix fail.

Adding that `SetValueIsSet()` fixes all of my test cases again.

Reply:
Indeed, good catch.

Fixed.